go/tools/builders/nogo_main.go

```diff
@@ -401,12 +407,30 @@ func checkAnalysisResults(actions []*action, pkg *goPackage) string {
 		}
 		config, ok := configs[act.a.Name]
 		if !ok {
-			// If the analyzer is not explicitly configured, it emits diagnostics for
-			// all files.
-			for _, diag := range act.diagnostics {
-				diagnostics = append(diagnostics, entry{Diagnostic: diag, Analyzer: act.a})
+			if baseConfig != nil {
+				config = *baseConfig
+				// Continue processing using the defaultConfig as config.
+			} else {
+				// If the analyzer is not explicitly configured, it emits diagnostics for
+				// all files.
+				for _, diag := range act.diagnostics {
+					diagnostics = append(diagnostics, entry{Diagnostic: diag, Analyzer: act.a})
+				}
+				continue
+			}
+			// If both the config AND the baseConfig exist, we try to merge them.
+			// If any fields in the analyzer config are unset, they will inherit
+			// the baseConfig config.
+		} else if baseConfig != nil {
+			if config.analyzerFlags == nil {
+				config.analyzerFlags = baseConfig.analyzerFlags
+			}
+			if config.onlyFiles == nil {
+				config.onlyFiles = baseConfig.onlyFiles
+			}
+			if config.excludeFiles == nil {
+				config.excludeFiles = baseConfig.excludeFiles
```

Comment:
```go
for _, act := range actions {
	// ....
	var currentConfig config

	if baseConfig, ok := configs[nogoBaseConfigName]; ok {
		currentConfig = baseConfig
	}

	if actionConfig, ok := configs[act.a.Name]; ok {
		if actionConfig.analyzerFlags != nil {
			currentConfig.analyzerFlags = actionConfig.analyzerFlags
		}
		if actionConfig.onlyFiles != nil {
			currentConfig.onlyFiles = actionConfig.onlyFiles
		}
		if actionConfig.excludeFiles != nil {
			currentConfig.excludeFiles = actionConfig.excludeFiles
		}
	}

	if currentConfig.onlyFiles == nil && currentConfig.excludeFiles == nil {
		for _, diag := range act.diagnostics {
			diagnostics = append(diagnostics, entry{Diagnostic: diag, Analyzer: act.a})
		}
		continue
	}

	//...
}
```

I found this easier to read: for each action, starting from an empty config, using base config if it exists, overriding it with action-specific config if exists. If it's still empty at this point, emit diagnostic for all files and continue to next action.

Reply:
Ah, nice idea! Applied!